node.go

```diff
@@ -206,6 +211,100 @@ func NewNode(srv *grpc.Server, cfg Config) (*Node, error) {
 	return n, nil
 }
 
+// NewHTTPNode creates an unstarted Node to participulate in a cluster. An error
+// will be returned if the provided config is invalid.
+func NewHTTPNode(mux *http.ServeMux, cfg Config) (*Node, error) {
+	if err := cfg.validate(); err != nil {
+		return nil, err
+	}
+
+	advertiseAddr, advertisePortString, err := net.SplitHostPort(cfg.AdvertiseAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read advertise address: %w", err)
+	}
+
+	advertiseIP, err := net.ResolveIPAddr("ip4", advertiseAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup advertise address %s: %w", advertiseAddr, err)
```

Comment:
Use `%q` in case the address is empty:

```suggestion
		return nil, fmt.Errorf("failed to lookup advertise address %q: %w", advertiseAddr, err)
```